torch/_inductor/compile_fx.py

```diff
@@ -693,7 +694,55 @@ def fw_compiler_base(model: torch.fx.GraphModule, example_inputs, is_inference):
         )
 
     fw_compiler = functools.partial(fw_compiler_base, is_inference=False)
-    inference_compiler = functools.partial(fw_compiler_base, is_inference=True)
+
+    if config.freezing:
+        from torch._inductor.freezing import freeze
+
+        def inference_compiler(
+            aot_autograd_model: torch.fx.GraphModule, example_inputs
+        ):
+            # partition_fn won't be called
+            joint_graph_passes(aot_autograd_model)
```

Comment:
It looks like before this PR, none of the joint_graph_passes would run when inductor sees an inference graph. Do we need to run them now for inference, but only when using `config.freezing`? (and more of a curiosity question - why?)

Reply:
oh i see, `fw_compiler_base()` calls the joint passes during inference. I guess you can't easily re-use it, since you wnat to run the joint passes before doing the freezing, but call `inner_compile` after the freezing.

And (answering my own question) it looks like we're guaranteed that the joint passes will always run- it looks like `fw_compiler` here doesn't invoke the joint passes, but we're guaranteed in aot autograd that we'll be running the partition_fn in that case, which also runs the passees.